ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/TestXAResourceRecovery.java

```diff
@@ -30,21 +30,32 @@
 public class TestXAResourceRecovery implements XAResourceRecovery {
 
 	private static Stack<XAResource> resources = new Stack<XAResource>();
+	private static int count = 0;
 
 	public XAResource getXAResource() throws SQLException {
-		return resources.pop();
+		count--;
```

Comment:
We are testing that our recovery system is calling XAResourceRecovery correctly. What if if we called it more that three times so count would equal -1 and then the hasMoreResources() call would always return true after that.

If this scenario cannot happen in the test then you should provide more documentation.